// angular2/ts/src/common/forms/directives.js
define(["require", "exports", 'angular2/src/facade/lang', './directives/ng_control_name', './directives/ng_form_control', './directives/ng_model', './directives/ng_control_group', './directives/ng_form_model', './directives/ng_form', './directives/default_value_accessor', './directives/checkbox_value_accessor', './directives/number_value_accessor', './directives/radio_control_value_accessor', './directives/ng_control_status', './directives/select_control_value_accessor', './directives/validators', './directives/ng_control_name', './directives/ng_form_control', './directives/ng_model', './directives/ng_control_group', './directives/ng_form_model', './directives/ng_form', './directives/default_value_accessor', './directives/checkbox_value_accessor', './directives/radio_control_value_accessor', './directives/number_value_accessor', './directives/ng_control_status', './directives/select_control_value_accessor', './directives/validators', './directives/ng_control'], function (require, exports, lang_1, ng_control_name_1, ng_form_control_1, ng_model_1, ng_control_group_1, ng_form_model_1, ng_form_1, default_value_accessor_1, checkbox_value_accessor_1, number_value_accessor_1, radio_control_value_accessor_1, ng_control_status_1, select_control_value_accessor_1, validators_1, ng_control_name_2, ng_form_control_2, ng_model_2, ng_control_group_2, ng_form_model_2, ng_form_2, default_value_accessor_2, checkbox_value_accessor_2, radio_control_value_accessor_2, number_value_accessor_2, ng_control_status_2, select_control_value_accessor_2, validators_2, ng_control_1) {
    "use strict";
    exports.NgControlName = ng_control_name_2.NgControlName;
    exports.NgFormControl = ng_form_control_2.NgFormControl;
    exports.NgModel = ng_model_2.NgModel;
    exports.NgControlGroup = ng_control_group_2.NgControlGroup;
    exports.NgFormModel = ng_form_model_2.NgFormModel;
    exports.NgForm = ng_form_2.NgForm;
    exports.DefaultValueAccessor = default_value_accessor_2.DefaultValueAccessor;
    exports.CheckboxControlValueAccessor = checkbox_value_accessor_2.CheckboxControlValueAccessor;
    exports.RadioControlValueAccessor = radio_control_value_accessor_2.RadioControlValueAccessor;
    exports.RadioButtonState = radio_control_value_accessor_2.RadioButtonState;
    exports.NumberValueAccessor = number_value_accessor_2.NumberValueAccessor;
    exports.NgControlStatus = ng_control_status_2.NgControlStatus;
    exports.SelectControlValueAccessor = select_control_value_accessor_2.SelectControlValueAccessor;
    exports.NgSelectOption = select_control_value_accessor_2.NgSelectOption;
    exports.RequiredValidator = validators_2.RequiredValidator;
    exports.MinLengthValidator = validators_2.MinLengthValidator;
    exports.MaxLengthValidator = validators_2.MaxLengthValidator;
    exports.PatternValidator = validators_2.PatternValidator;
    exports.NgControl = ng_control_1.NgControl;
    /**
     *
     * A list of all the form directives used as part of a `@View` annotation.
     *
     *  This is a shorthand for importing them each individually.
     *
     * ### Example
     *
     * ```typescript
     * @Component({
     *   selector: 'my-app',
     *   directives: [FORM_DIRECTIVES]
     * })
     * class MyApp {}
     * ```
     */
    exports.FORM_DIRECTIVES = lang_1.CONST_EXPR([
        ng_control_name_1.NgControlName,
        ng_control_group_1.NgControlGroup,
        ng_form_control_1.NgFormControl,
        ng_model_1.NgModel,
        ng_form_model_1.NgFormModel,
        ng_form_1.NgForm,
        select_control_value_accessor_1.NgSelectOption,
        default_value_accessor_1.DefaultValueAccessor,
        number_value_accessor_1.NumberValueAccessor,
        checkbox_value_accessor_1.CheckboxControlValueAccessor,
        select_control_value_accessor_1.SelectControlValueAccessor,
        radio_control_value_accessor_1.RadioControlValueAccessor,
        ng_control_status_1.NgControlStatus,
        validators_1.RequiredValidator,
        validators_1.MinLengthValidator,
        validators_1.MaxLengthValidator,
        validators_1.PatternValidator
    ]);
});
