// angular2/ts/src/compiler/compiler.js
define(["require", "exports", "./runtime_compiler", './template_compiler', './directive_metadata', './source_module', 'angular2/src/core/platform_directives_and_pipes', 'angular2/src/compiler/template_ast', 'angular2/src/compiler/template_parser', 'angular2/src/facade/lang', 'angular2/src/core/di', 'angular2/src/compiler/template_parser', 'angular2/src/compiler/html_parser', 'angular2/src/compiler/template_normalizer', 'angular2/src/compiler/runtime_metadata', 'angular2/src/compiler/change_detector_compiler', 'angular2/src/compiler/style_compiler', 'angular2/src/compiler/view_compiler', 'angular2/src/compiler/proto_view_compiler', 'angular2/src/compiler/template_compiler', 'angular2/src/core/change_detection/change_detection', 'angular2/src/core/linker/compiler', 'angular2/src/compiler/runtime_compiler', 'angular2/src/compiler/schema/element_schema_registry', 'angular2/src/compiler/schema/dom_element_schema_registry', 'angular2/src/compiler/url_resolver', 'angular2/src/core/change_detection/change_detection'], function (require, exports, runtime_compiler_1, template_compiler_1, directive_metadata_1, source_module_1, platform_directives_and_pipes_1, template_ast_1, template_parser_1, lang_1, di_1, template_parser_2, html_parser_1, template_normalizer_1, runtime_metadata_1, change_detector_compiler_1, style_compiler_1, view_compiler_1, proto_view_compiler_1, template_compiler_2, change_detection_1, compiler_1, runtime_compiler_2, element_schema_registry_1, dom_element_schema_registry_1, url_resolver_1, change_detection_2) {
    "use strict";
    function __export(m) {
        for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
    }
    exports.TemplateCompiler = template_compiler_1.TemplateCompiler;
    exports.CompileDirectiveMetadata = directive_metadata_1.CompileDirectiveMetadata;
    exports.CompileTypeMetadata = directive_metadata_1.CompileTypeMetadata;
    exports.CompileTemplateMetadata = directive_metadata_1.CompileTemplateMetadata;
    exports.SourceModule = source_module_1.SourceModule;
    exports.SourceWithImports = source_module_1.SourceWithImports;
    exports.PLATFORM_DIRECTIVES = platform_directives_and_pipes_1.PLATFORM_DIRECTIVES;
    exports.PLATFORM_PIPES = platform_directives_and_pipes_1.PLATFORM_PIPES;
    __export(template_ast_1);
    exports.TEMPLATE_TRANSFORMS = template_parser_1.TEMPLATE_TRANSFORMS;
    function _createChangeDetectorGenConfig() {
        return new change_detection_1.ChangeDetectorGenConfig(lang_1.assertionsEnabled(), false, true);
    }
    /**
     * A set of providers that provide `RuntimeCompiler` and its dependencies to use for
     * template compilation.
     */
    exports.COMPILER_PROVIDERS = lang_1.CONST_EXPR([
        change_detection_2.Lexer,
        change_detection_2.Parser,
        html_parser_1.HtmlParser,
        template_parser_2.TemplateParser,
        template_normalizer_1.TemplateNormalizer,
        runtime_metadata_1.RuntimeMetadataResolver,
        url_resolver_1.DEFAULT_PACKAGE_URL_PROVIDER,
        style_compiler_1.StyleCompiler,
        proto_view_compiler_1.ProtoViewCompiler,
        view_compiler_1.ViewCompiler,
        change_detector_compiler_1.ChangeDetectionCompiler,
        new di_1.Provider(change_detection_1.ChangeDetectorGenConfig, { useFactory: _createChangeDetectorGenConfig, deps: [] }),
        template_compiler_2.TemplateCompiler,
        new di_1.Provider(runtime_compiler_2.RuntimeCompiler, { useClass: runtime_compiler_1.RuntimeCompiler_ }),
        new di_1.Provider(compiler_1.Compiler, { useExisting: runtime_compiler_2.RuntimeCompiler }),
        dom_element_schema_registry_1.DomElementSchemaRegistry,
        new di_1.Provider(element_schema_registry_1.ElementSchemaRegistry, { useExisting: dom_element_schema_registry_1.DomElementSchemaRegistry }),
        url_resolver_1.UrlResolver
    ]);
});
