// angular2/ts/platform/common_dom.js
define(["require", "exports", 'angular2/src/platform/dom/dom_adapter', 'angular2/src/platform/dom/dom_renderer', 'angular2/src/platform/dom/dom_tokens', 'angular2/src/platform/dom/shared_styles_host', 'angular2/src/platform/dom/events/dom_events', 'angular2/src/platform/dom/events/event_manager', 'angular2/src/platform/dom/debug/by', 'angular2/src/platform/dom/debug/ng_probe'], function (require, exports, dom_adapter_1, dom_renderer_1, dom_tokens_1, shared_styles_host_1, dom_events_1, event_manager_1, by_1, ng_probe_1) {
    "use strict";
    function __export(m) {
        for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
    }
    /**
     * This is a set of classes and objects that can be used both in the browser and on the server.
     */
    exports.DOM = dom_adapter_1.DOM;
    exports.setRootDomAdapter = dom_adapter_1.setRootDomAdapter;
    exports.DomAdapter = dom_adapter_1.DomAdapter;
    exports.DomRenderer = dom_renderer_1.DomRenderer;
    exports.DOCUMENT = dom_tokens_1.DOCUMENT;
    exports.SharedStylesHost = shared_styles_host_1.SharedStylesHost;
    exports.DomSharedStylesHost = shared_styles_host_1.DomSharedStylesHost;
    exports.DomEventsPlugin = dom_events_1.DomEventsPlugin;
    exports.EVENT_MANAGER_PLUGINS = event_manager_1.EVENT_MANAGER_PLUGINS;
    exports.EventManager = event_manager_1.EventManager;
    exports.EventManagerPlugin = event_manager_1.EventManagerPlugin;
    __export(by_1);
    __export(ng_probe_1);
});
