// angular2/ts/platform/worker_app.js
define(["require", "exports", 'angular2/src/platform/worker_app_common', 'angular2/src/platform/worker_app', 'angular2/src/web_workers/shared/client_message_broker', 'angular2/src/web_workers/shared/service_message_broker', 'angular2/src/web_workers/shared/serializer', 'angular2/src/web_workers/shared/message_bus', 'angular2/src/core/angular_entrypoint', 'angular2/src/web_workers/worker/router_providers'], function (require, exports, worker_app_common_1, worker_app_1, client_message_broker_1, service_message_broker_1, serializer_1, message_bus_1, angular_entrypoint_1, router_providers_1) {
    "use strict";
    function __export(m) {
        for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
    }
    exports.WORKER_APP_PLATFORM = worker_app_common_1.WORKER_APP_PLATFORM;
    exports.WORKER_APP_APPLICATION_COMMON = worker_app_common_1.WORKER_APP_APPLICATION_COMMON;
    exports.WORKER_APP_APPLICATION = worker_app_1.WORKER_APP_APPLICATION;
    exports.ClientMessageBroker = client_message_broker_1.ClientMessageBroker;
    exports.ClientMessageBrokerFactory = client_message_broker_1.ClientMessageBrokerFactory;
    exports.FnArg = client_message_broker_1.FnArg;
    exports.UiArguments = client_message_broker_1.UiArguments;
    exports.ReceivedMessage = service_message_broker_1.ReceivedMessage;
    exports.ServiceMessageBroker = service_message_broker_1.ServiceMessageBroker;
    exports.ServiceMessageBrokerFactory = service_message_broker_1.ServiceMessageBrokerFactory;
    exports.PRIMITIVE = serializer_1.PRIMITIVE;
    __export(message_bus_1);
    exports.AngularEntrypoint = angular_entrypoint_1.AngularEntrypoint;
    exports.WORKER_APP_ROUTER = router_providers_1.WORKER_APP_ROUTER;
});
