// angular2/ts/platform/worker_render.js
define(["require", "exports", 'angular2/src/platform/worker_render_common', 'angular2/src/platform/worker_render', '../src/web_workers/shared/client_message_broker', '../src/web_workers/shared/service_message_broker', '../src/web_workers/shared/serializer', '../src/web_workers/shared/message_bus', 'angular2/src/platform/worker_render', 'angular2/src/web_workers/ui/router_providers'], function (require, exports, worker_render_common_1, worker_render_1, client_message_broker_1, service_message_broker_1, serializer_1, message_bus_1, worker_render_2, router_providers_1) {
    "use strict";
    function __export(m) {
        for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
    }
    exports.WORKER_SCRIPT = worker_render_common_1.WORKER_SCRIPT;
    exports.WORKER_RENDER_PLATFORM = worker_render_common_1.WORKER_RENDER_PLATFORM;
    exports.initializeGenericWorkerRenderer = worker_render_common_1.initializeGenericWorkerRenderer;
    exports.WORKER_RENDER_APPLICATION_COMMON = worker_render_common_1.WORKER_RENDER_APPLICATION_COMMON;
    exports.WORKER_RENDER_APPLICATION = worker_render_1.WORKER_RENDER_APPLICATION;
    exports.WebWorkerInstance = worker_render_1.WebWorkerInstance;
    exports.ClientMessageBroker = client_message_broker_1.ClientMessageBroker;
    exports.ClientMessageBrokerFactory = client_message_broker_1.ClientMessageBrokerFactory;
    exports.FnArg = client_message_broker_1.FnArg;
    exports.UiArguments = client_message_broker_1.UiArguments;
    exports.ReceivedMessage = service_message_broker_1.ReceivedMessage;
    exports.ServiceMessageBroker = service_message_broker_1.ServiceMessageBroker;
    exports.ServiceMessageBrokerFactory = service_message_broker_1.ServiceMessageBrokerFactory;
    exports.PRIMITIVE = serializer_1.PRIMITIVE;
    __export(message_bus_1);
    /**
     * @deprecated Use WORKER_RENDER_APPLICATION
     */
    exports.WORKER_RENDER_APP = worker_render_2.WORKER_RENDER_APPLICATION;
    exports.WORKER_RENDER_ROUTER = router_providers_1.WORKER_RENDER_ROUTER;
});
