// angular2/ts/src/core/linker.js
define(["require", "exports", './linker/directive_resolver', './linker/view_resolver', './linker/compiler', './linker/view_manager', './linker/query_list', './linker/dynamic_component_loader', './linker/element_ref', './linker/template_ref', './linker/view_ref', './linker/view_container_ref', './linker/dynamic_component_loader'], function (require, exports, directive_resolver_1, view_resolver_1, compiler_1, view_manager_1, query_list_1, dynamic_component_loader_1, element_ref_1, template_ref_1, view_ref_1, view_container_ref_1, dynamic_component_loader_2) {
    "use strict";
    // Public API for compiler
    exports.DirectiveResolver = directive_resolver_1.DirectiveResolver;
    exports.ViewResolver = view_resolver_1.ViewResolver;
    exports.Compiler = compiler_1.Compiler;
    exports.AppViewManager = view_manager_1.AppViewManager;
    exports.QueryList = query_list_1.QueryList;
    exports.DynamicComponentLoader = dynamic_component_loader_1.DynamicComponentLoader;
    exports.ElementRef = element_ref_1.ElementRef;
    exports.TemplateRef = template_ref_1.TemplateRef;
    exports.EmbeddedViewRef = view_ref_1.EmbeddedViewRef;
    exports.HostViewRef = view_ref_1.HostViewRef;
    exports.ViewRef = view_ref_1.ViewRef;
    exports.HostViewFactoryRef = view_ref_1.HostViewFactoryRef;
    exports.ViewContainerRef = view_container_ref_1.ViewContainerRef;
    exports.ComponentRef = dynamic_component_loader_2.ComponentRef;
});
