// angular2/ts/core.js
define(["require", "exports", './src/core/metadata', './src/core/util', './src/core/prod_mode', './src/core/di', './src/facade/facade', 'angular2/src/facade/lang', './src/core/application_ref', './src/core/application_tokens', './src/core/zone', './src/core/render', './src/core/linker', './src/core/debug/debug_node', './src/core/testability/testability', './src/core/change_detection', './src/core/platform_directives_and_pipes', './src/core/platform_common_providers', './src/core/application_common_providers', './src/core/reflection/reflection'], function (require, exports, metadata_1, util_1, prod_mode_1, di_1, facade_1, lang_1, application_ref_1, application_tokens_1, zone_1, render_1, linker_1, debug_node_1, testability_1, change_detection_1, platform_directives_and_pipes_1, platform_common_providers_1, application_common_providers_1, reflection_1) {
    "use strict";
    function __export(m) {
        for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
    }
    /**
     * @module
     * @description
     * Starting point to import all public core APIs.
     */
    __export(metadata_1);
    __export(util_1);
    __export(prod_mode_1);
    __export(di_1);
    __export(facade_1);
    exports.enableProdMode = lang_1.enableProdMode;
    exports.platform = application_ref_1.platform;
    exports.createNgZone = application_ref_1.createNgZone;
    exports.PlatformRef = application_ref_1.PlatformRef;
    exports.ApplicationRef = application_ref_1.ApplicationRef;
    exports.APP_ID = application_tokens_1.APP_ID;
    exports.APP_COMPONENT = application_tokens_1.APP_COMPONENT;
    exports.APP_INITIALIZER = application_tokens_1.APP_INITIALIZER;
    exports.PACKAGE_ROOT_URL = application_tokens_1.PACKAGE_ROOT_URL;
    exports.PLATFORM_INITIALIZER = application_tokens_1.PLATFORM_INITIALIZER;
    __export(zone_1);
    __export(render_1);
    __export(linker_1);
    exports.DebugElement = debug_node_1.DebugElement;
    exports.DebugNode = debug_node_1.DebugNode;
    exports.asNativeElements = debug_node_1.asNativeElements;
    __export(testability_1);
    __export(change_detection_1);
    __export(platform_directives_and_pipes_1);
    __export(platform_common_providers_1);
    __export(application_common_providers_1);
    __export(reflection_1);
});
