// angular2/ts/router.js
/**
 * @module
 * @description
 * Maps application URLs into application states, to support deep-linking and navigation.
 */
define(["require", "exports", './src/router/router', './src/router/router_outlet', './src/router/router_link', './src/router/instruction', './src/router/platform_location', './src/router/route_registry', './src/router/location_strategy', './src/router/hash_location_strategy', './src/router/path_location_strategy', './src/router/location', './src/router/route_config_decorator', './src/router/lifecycle_annotations', './src/router/instruction', 'angular2/core', 'angular2/src/router/router_providers_common', 'angular2/src/router/router_providers', './src/router/router_outlet', './src/router/router_link', './src/facade/lang'], function (require, exports, router_1, router_outlet_1, router_link_1, instruction_1, platform_location_1, route_registry_1, location_strategy_1, hash_location_strategy_1, path_location_strategy_1, location_1, route_config_decorator_1, lifecycle_annotations_1, instruction_2, core_1, router_providers_common_1, router_providers_1, router_outlet_2, router_link_2, lang_1) {
    "use strict";
    function __export(m) {
        for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
    }
    exports.Router = router_1.Router;
    exports.RouterOutlet = router_outlet_1.RouterOutlet;
    exports.RouterLink = router_link_1.RouterLink;
    exports.RouteParams = instruction_1.RouteParams;
    exports.RouteData = instruction_1.RouteData;
    exports.PlatformLocation = platform_location_1.PlatformLocation;
    exports.RouteRegistry = route_registry_1.RouteRegistry;
    exports.ROUTER_PRIMARY_COMPONENT = route_registry_1.ROUTER_PRIMARY_COMPONENT;
    exports.LocationStrategy = location_strategy_1.LocationStrategy;
    exports.APP_BASE_HREF = location_strategy_1.APP_BASE_HREF;
    exports.HashLocationStrategy = hash_location_strategy_1.HashLocationStrategy;
    exports.PathLocationStrategy = path_location_strategy_1.PathLocationStrategy;
    exports.Location = location_1.Location;
    __export(route_config_decorator_1);
    exports.CanActivate = lifecycle_annotations_1.CanActivate;
    exports.Instruction = instruction_2.Instruction;
    exports.ComponentInstruction = instruction_2.ComponentInstruction;
    exports.OpaqueToken = core_1.OpaqueToken;
    exports.ROUTER_PROVIDERS_COMMON = router_providers_common_1.ROUTER_PROVIDERS_COMMON;
    exports.ROUTER_PROVIDERS = router_providers_1.ROUTER_PROVIDERS;
    exports.ROUTER_BINDINGS = router_providers_1.ROUTER_BINDINGS;
    /**
     * A list of directives. To use the router directives like {@link RouterOutlet} and
     * {@link RouterLink}, add this to your `directives` array in the {@link View} decorator of your
     * component.
     *
     * ### Example ([live demo](http://plnkr.co/edit/iRUP8B5OUbxCWQ3AcIDm))
     *
     * ```
     * import {Component} from 'angular2/core';
     * import {ROUTER_DIRECTIVES, ROUTER_PROVIDERS, RouteConfig} from 'angular2/router';
     *
     * @Component({directives: [ROUTER_DIRECTIVES]})
     * @RouteConfig([
     *  {...},
     * ])
     * class AppCmp {
     *    // ...
     * }
     *
     * bootstrap(AppCmp, [ROUTER_PROVIDERS]);
     * ```
     */
    exports.ROUTER_DIRECTIVES = lang_1.CONST_EXPR([router_outlet_2.RouterOutlet, router_link_2.RouterLink]);
});
