// angular2/ts/testing.js
define(["require", "exports", './src/testing/testing', './src/testing/test_component_builder', './src/testing/test_injector', './src/testing/fake_async', 'angular2/src/mock/view_resolver_mock', 'angular2/src/compiler/xhr_mock', 'angular2/src/mock/ng_zone_mock', 'angular2/src/mock/mock_application_ref', 'angular2/src/mock/directive_resolver_mock'], function (require, exports, testing_1, test_component_builder_1, test_injector_1, fake_async_1, view_resolver_mock_1, xhr_mock_1, ng_zone_mock_1, mock_application_ref_1, directive_resolver_mock_1) {
    "use strict";
    function __export(m) {
        for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
    }
    /**
     * @module
     * @description
     * This module is used for writing tests for applications written in Angular.
     *
     * This module is not included in the `angular2` module; you must import the test module explicitly.
     *
     */
    __export(testing_1);
    exports.ComponentFixture = test_component_builder_1.ComponentFixture;
    exports.TestComponentBuilder = test_component_builder_1.TestComponentBuilder;
    __export(test_injector_1);
    __export(fake_async_1);
    exports.MockViewResolver = view_resolver_mock_1.MockViewResolver;
    exports.MockXHR = xhr_mock_1.MockXHR;
    exports.MockNgZone = ng_zone_mock_1.MockNgZone;
    exports.MockApplicationRef = mock_application_ref_1.MockApplicationRef;
    exports.MockDirectiveResolver = directive_resolver_mock_1.MockDirectiveResolver;
});
