// angular2/ts/src/common/pipes.js
/**
 * @module
 * @description
 * This module provides a set of common Pipes.
 */
define(["require", "exports", './pipes/async_pipe', './pipes/date_pipe', './pipes/json_pipe', './pipes/slice_pipe', './pipes/lowercase_pipe', './pipes/number_pipe', './pipes/uppercase_pipe', './pipes/replace_pipe', './pipes/i18n_plural_pipe', './pipes/i18n_select_pipe', './pipes/common_pipes'], function (require, exports, async_pipe_1, date_pipe_1, json_pipe_1, slice_pipe_1, lowercase_pipe_1, number_pipe_1, uppercase_pipe_1, replace_pipe_1, i18n_plural_pipe_1, i18n_select_pipe_1, common_pipes_1) {
    "use strict";
    exports.AsyncPipe = async_pipe_1.AsyncPipe;
    exports.DatePipe = date_pipe_1.DatePipe;
    exports.JsonPipe = json_pipe_1.JsonPipe;
    exports.SlicePipe = slice_pipe_1.SlicePipe;
    exports.LowerCasePipe = lowercase_pipe_1.LowerCasePipe;
    exports.NumberPipe = number_pipe_1.NumberPipe;
    exports.DecimalPipe = number_pipe_1.DecimalPipe;
    exports.PercentPipe = number_pipe_1.PercentPipe;
    exports.CurrencyPipe = number_pipe_1.CurrencyPipe;
    exports.UpperCasePipe = uppercase_pipe_1.UpperCasePipe;
    exports.ReplacePipe = replace_pipe_1.ReplacePipe;
    exports.I18nPluralPipe = i18n_plural_pipe_1.I18nPluralPipe;
    exports.I18nSelectPipe = i18n_select_pipe_1.I18nSelectPipe;
    exports.COMMON_PIPES = common_pipes_1.COMMON_PIPES;
});
